x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/crossclusteraccess/CrossClusterAccessAuthenticationServiceIntegTests.java

```diff
@@ -124,15 +119,18 @@ public void testInvalidHeaders() throws IOException {
         try (var ignored = threadContext.stashContext()) {
             new CrossClusterAccessHeaders(
                 encodedCrossClusterAccessApiKey,
-                AuthenticationTestHelper.randomCrossClusterAccessSubjectInfo(
-                    new RoleDescriptorsIntersection(
-                        randomValueOtherThanMany(
-                            rd -> false == (rd.hasClusterPrivileges()
-                                || rd.hasApplicationPrivileges()
-                                || rd.hasConfigurableClusterPrivileges()
-                                || rd.hasRunAs()
-                                || rd.hasRemoteIndicesPrivileges()),
-                            () -> RoleDescriptorTests.randomRoleDescriptor()
+                randomValueOtherThanMany(
+                    cca -> CrossClusterAccessUser.is(cca.getAuthentication().getEffectiveSubject().getUser()),
+                    () -> AuthenticationTestHelper.randomCrossClusterAccessSubjectInfo(
+                        new RoleDescriptorsIntersection(
```

Comment:
IIUC, `randomCrossClusterAccessSubjectInfo(RoleDescriptorsIntersection)` should never return an internal user so that the `otherThanMany` check is redundant.

Reply:
Indeed.